x-pack/filebeat/input/http_endpoint/input.go

```diff
@@ -82,16 +82,25 @@ func (e *httpEndpoint) Test(_ v2.TestContext) error {
 func (e *httpEndpoint) Run(ctx v2.Context, publisher stateless.Publisher) error {
 	log := ctx.Logger.With("address", e.addr)
 
-	validator := &apiValidator{
+	authValidator := &authValidator{
 		basicAuth:    e.config.BasicAuth,
 		username:     e.config.Username,
 		password:     e.config.Password,
-		method:       http.MethodPost,
-		contentType:  e.config.ContentType,
 		secretHeader: e.config.SecretHeader,
 		secretValue:  e.config.SecretValue,
 	}
 
+	hmacValidator := &hmacValidator{
+		hmacHeader: e.config.HmacHeader,
+		hmacPrefix: e.config.HmacPrefix,
+		hmacToken:  e.config.HmacToken,
+	}
+
+	headerValidator := &headerValidator{
+		method:      http.MethodPost,
+		contentType: e.config.ContentType,
+	}
```

Comment:
The validators need to check on every request if they are configured or not. Worse, they even check for missconfiguration at runtime (users can detect beats miss configuration in HTTP client at runtime only). I'd propose to separate configuration time from runtime, and build an array of configured/enabled header validators (e.g. `validatorList` type shown in another comment). Faulty configurations must be detected when creating the endpoint object, not maybe later after running it.